main.tf

```diff
@@ -14,8 +14,173 @@
  * limitations under the License.
  */
 
-resource "google_storage_bucket" "main" {
-  project  = var.project_id
-  name     = var.bucket_name
-  location = "US"
+data "google_project" "project" {
+  project_id = var.project
 }
+
+
+locals {
+
+
+  trigger_sa = compact([var.cloud_trigger_sa == null ? "" : join("=>", [var.project, var.cloud_trigger_sa])])
+
+  target_sa = compact(distinct(flatten([for j in var.stage_targets : j.execution_configs_service_account != null ? join("=>", [element(split("/", j.gke), 1), j.execution_configs_service_account, var.project]) : ""])))
```

Comment:
Can we look for a simpler solution to this method of parsing?

Reply:
This has been addressed. Please check